### PR TITLE
Add ruff rules for bandit

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: examples/notebooks/conftest.py [hunk 1]
```diff
@@ -16,9 +17,9 @@ def get_required_env(name) -> str:
 
 # vertex-ai
 if "GOOGLE_APPLICATION_CREDENTIALS" not in os.environ:
-    with open("/tmp/gcloud-account-key.json", "w") as f:
+    with tempfile.NamedTemporaryFile(delete=False) as f:
```
Comment (cbornet) [inline examples/notebooks/conftest.py:20]: Fix S108: https://docs.astral.sh/ruff/rules/hardcoded-temp-file/

File: docker/examples/basic/app.py [hunk 2]
```diff
@@ -32,4 +32,4 @@ def hello():
 
 # Execute the application
 if __name__ == "__main__":
-    app.run(port=8080, host="0.0.0.0")
+    app.run(port=8080)
```
Comment (cbornet) [inline docker/examples/basic/app.py:35]: Fix S104: https://docs.astral.sh/ruff/rules/hardcoded-bind-all-interfaces/

File: pyproject.toml [hunk 3]
```diff
@@ -60,6 +60,7 @@ extend-include = ["*.ipynb"]
 [tool.ruff.lint]
 ignore = [
     "PERF203", # Incorrect detection
+    "S608", # TODO: fix possible injections in CQL ?
```
Comment (cbornet) [inline pyproject.toml:63]: Should we sanitize keyspace, node_table, targets_table before passing to query ?

Comment (cbornet) [reply pyproject.toml]: Added sanitization.

**Review by bjchambers — APPROVED** [review]

File: libs/e2e-tests/e2e_tests/conftest.py [hunk 4]
```diff
@@ -199,6 +200,6 @@ def _report_to_file(stats_str: str, filename: str, report_lines: list):
 
 # vertex-ai
 if "GOOGLE_APPLICATION_CREDENTIALS" not in os.environ:
-    with open("/tmp/gcloud-account-key.json", "w") as f:
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as f:
```
[the remark this answers is not in the document]
Comment (cbornet) [reply libs/e2e-tests/e2e_tests/conftest.py]: done

**Pull request merged**